pkg/common/k8s_utils_test.go

```diff
@@ -359,3 +371,527 @@ func TestIsObjectTaggedToDelete(t *testing.T) {
+							Name:       "http2",
+							Port:       8090,
+							TargetPort: intstr.FromInt(8090),
+						},
+						{
+							Name:       "http3",
+							Port:       8100,
+							TargetPort: intstr.FromInt(8100),
+						},
+					},
+				},
+			},
+			serviceKey: client.ObjectKey{Name: "my-service6", Namespace: "default"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.service != nil {
+				err := k8sClient.Create(ctx, tt.service)
+				if err != nil {
+					t.Fatalf("failed to create service: %v", err)
+				}
+			}
+
+			portNumber, err := GetServicePortNumber(ctx, k8sClient, tt.serviceKey, tt.servicePort)
+
+			if err != nil && tt.expectedErr == nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+			if err == nil && tt.expectedErr != nil {
+				t.Error("expected an error, but got nil")
+			}
+			if err != nil && tt.expectedErr != nil && err.Error() != tt.expectedErr.Error() {
+				t.Errorf("unexpected error: got %v, want %v", err, tt.expectedErr)
+			}
+
+			if portNumber != tt.expected {
+				t.Errorf("unexpected port number: got %d, want %d", portNumber, tt.expected)
+			}
+		})
+	}
+}
+
+func TestContainsObjectKey(t *testing.T) {
+	key1 := client.ObjectKey{Namespace: "ns1", Name: "obj1"}
+	key2 := client.ObjectKey{Namespace: "ns2", Name: "obj2"}
+	key3 := client.ObjectKey{Namespace: "ns3", Name: "obj3"}
+	key4 := client.ObjectKey{Namespace: "ns4", Name: "obj4"}
+
+	testCases := []struct {
+		name     string
+		list     []client.ObjectKey
+		key      client.ObjectKey
+		expected bool
+	}{
+		{
+			name:     "when list contains key then return true",
+			list:     []client.ObjectKey{key1, key2, key3},
+			key:      key2,
+			expected: true,
+		},
+		{
+			name:     "when list does not contain key then return false",
+			list:     []client.ObjectKey{key1, key2, key3},
+			key:      key4,
+			expected: false,
+		},
+		{
+			name:     "when list is empty then return false",
+			list:     []client.ObjectKey{},
+			key:      key4,
+			expected: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := ContainsObjectKey(tc.list, tc.key)
+			if result != tc.expected {
+				t.Errorf("unexpected result: got %t, want %t", result, tc.expected)
+			}
+		})
+	}
+}
+
+func TestFindObjectKey(t *testing.T) {
+	key1 := client.ObjectKey{Namespace: "ns1", Name: "obj1"}
+	key2 := client.ObjectKey{Namespace: "ns2", Name: "obj2"}
+	key3 := client.ObjectKey{Namespace: "ns3", Name: "obj3"}
+
+	testCases := []struct {
+		name     string
+		list     []client.ObjectKey
+		key      client.ObjectKey
+		expected int
+	}{
+		{
+			name:     "when input slice has one search ObjectKey then return its index",
+			list:     []client.ObjectKey{key1, key2, key3},
+			key:      key2,
+			expected: 1,
+		},
+		{
+			name:     "when input slice has no search ObjectKey then return length of input slice",
+			list:     []client.ObjectKey{key1, key3},
+			key:      key2,
+			expected: 2,
+		},
```

Comment:
No problem with the test itself, but I'd probably expect a classical index-of kind of behavior here, where the returned value is `-1` when not found, instead of the length of the input array.

Reply:
I totally agre with you, Gui! Unfortunaletly, the len(...) implementation has already been used in other parts several times. So, there are situations where the implementation is tighted to checking the length of the slice, and such situations occur: `if Index != len(slice) ==> do something`

What do you think, @didierofrivia ?

Reply:
That function in particular does not make any sense in terms that when there are no objects found, it  will return the length (0 or more). One might want to check the usage of that function and see why that behaviour is useful.. marking it as resolved tho.

Reply:
I've removed this func from https://github.com/Kuadrant/gateway-api-machinery. Implementations can rely on [`common.IndexOf`](https://github.com/Kuadrant/gateway-api-machinery/blob/f89dbe5ac1de65202a92dd6c4c8afe5aca1639bd/common/slice_utils.go#L24-L33) instead.